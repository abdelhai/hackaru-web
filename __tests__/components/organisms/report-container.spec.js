import Factory from '@/__tests__/__setups__/factory';
import ReportContainer from '@/components/organisms/report-container';

describe('ReportContainer', () => {
  it('render correctly', () => {
    const wrapper = new Factory(ReportContainer, {
      propsData: {
        doughnutChartData: {
          labels: ['Development', 'Review'],
          datasets: [{ data: [100, 50], backgroundColor: ['#ff0', '#f00'] }]
        },
        barChartData: {
          labels: [
            'Jan',
            'Feb',
            'Mar',
            'Apr',
            'May',
            'Jun',
            'Jul',
            'Aug',
            'Sep',
            'Oct',
            'Nov',
            'Dec'
          ],
          datasets: [
            {
              label: 'Development',
              backgroundColor: '#ff0',
              data: [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10]
            },
            {
              label: 'Review',
              backgroundColor: '#ff0',
              data: [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10]
            }
          ]
        },
        summary: {
          1: 100,
          2: 50
        },
        projects: [
          {
            id: 1,
            name: 'Development',
            color: '#ff0'
          },
          {
            id: 2,
            name: 'Review',
            color: '#f00'
          }
        ]
      }
    }).shallow();
    expect(wrapper.element).toMatchSnapshot();
  });
});
