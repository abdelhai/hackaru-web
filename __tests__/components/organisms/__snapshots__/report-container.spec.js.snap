// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportContainer render correctly 1`] = `
<article
  class="report-container"
>
  <div
    class="bar-chart-wrapper"
  >
    <bar-chart-stub
      chartdata="[object Object]"
      class="bar-chart"
      options="[object Object]"
    />
  </div>
   
  <div
    class="content"
  >
    <div
      class="doughnut-chart-wrapper"
    >
      <!---->
       
      <doughnut-chart-stub
        chartdata="[object Object]"
        class="doughnut-chart"
        options="[object Object]"
      />
    </div>
     
    <ul
      class="projects"
    >
      <li>
        <project-name-stub
          color="#ff0"
          name="Development"
        />
         
        <time
          class="duration"
        >
          
          00:01:40
        
        </time>
      </li>
      <li>
        <project-name-stub
          color="#f00"
          name="Review"
        />
         
        <time
          class="duration"
        >
          
          00:00:50
        
        </time>
      </li>
    </ul>
  </div>
</article>
`;
